src/Plugins/ComplexCore/src/ComplexCore/Filters/RemoveFlaggedVertices.cpp

```diff
@@ -133,11 +137,17 @@ IFilter::PreflightResult RemoveFlaggedVertices::preflightImpl(const DataStructur
 
   for(const auto& targetArrayPath : targetArrayPaths)
   {
-    auto* targetArray = dataStructure.getDataAs<IDataArray>(targetArrayPath);
-    if(!targetArray)
+    const auto* targetArray = dataStructure.getDataAs<IDataArray>(targetArrayPath);
+    if(targetArray == nullptr)
     {
-      std::string ss = fmt::format("Array not found at path: '{}'", targetArrayPath.toString());
-      return {MakeErrorResult<OutputActions>(::k_ArrayNotFound, ss)};
+      std::string errorMsg = fmt::format("Array not found at path: '{}'", targetArrayPath.toString());
+      return {MakeErrorResult<OutputActions>(::k_ArrayNotFound, errorMsg)};
+    }
+    // Make sure selected arrays are 1D Arrays (number of components does not matter)
+    if(targetArray->getIDataStore()->getTupleShape().size() != 1)
```

Comment:
```suggestion
    if(targetArray->getIDataStoreRef().getTupleShape().size() != 1)
```

Reply:
What is the real difference between getting the pointer and getting the reference? Is there a performance difference?

Reply:
Safety. We'll throw an exception rather than have potential undefined behavior. Generally speaking a reference is preferable to a pointer and should be the first choice. Pointers should only be used if there's a specific need for it.